connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -284,7 +312,7 @@ public void halt() {
             }
             for (ConnectorTaskId taskId : new HashSet<>(worker.taskIds())) {
                 try {
-                    worker.stopTask(taskId);
+                    worker.stopAndAwaitTask(taskId);
```

Comment:
You added `Worker.stopTasks` and `Worker.awaitStopTasks`. Any reason not to convert this code to use those methods (and allow the tasks to stop in parallel) instead of handling them serially like this?